examples/kubeflow/mnist/pipeline.py

```diff
@@ -82,21 +88,124 @@ def kubeflow_pipeline(s3_endpoint: str, input_bucket: str):
 
     return op(s3_endpoint, input_bucket)
 
-def main():
-    client = kfp.Client()
-
-    run_id = client.create_run_from_pipeline_func(
-        kubeflow_pipeline,
-        arguments={
-            "s3_endpoint": os.environ["S3_ENDPOINT"],
-            "input_bucket": "input",
-        }
-    ).run_id
+parser = argparse.ArgumentParser(description='Create a kubeflow pipeline ' +
+    'that trains a small neural net on MNIST data')
+parser.add_argument('--remote_host',
+                    default="", metavar="http://host/path", type=str,
+                    help='The address of the kubeflow pipelines API, if '+
+                         'running outside a kubeflow cluster')
+parser.add_argument('--create_run_in',
+                    default="", metavar="pipeline_name", type=str,
+                    help='If set, this script will create a run for the '+
+                    'given kubeflow pipeline and wait for it to complete, '+
+                    'rather than uploading a new pipeline')
+parser.add_argument('--create_pipeline',
+                    default="", metavar="pipeline_name", type=str,
+                    help='If set, this script will create a kubeflow pipeline '+
+                    'with the given name, but will not create a run in that '+
+                    'pipeline. Used to create a pipeline from a local machine.')
+parser.add_argument('--force', action='store_true', default=False,
+                    help='If set, and this script tries to create a pipeline '+
+                    'where a kubeflow pipeline with the same name already '+
+                    'exists, the existing pipeline will be deleted')
```

Comment:
1) Put `parser` either in the `if __name__ == '__main__` block, and pass the arguments into the main method. This avoids globals and makes it easier in the future to hook into main outside of the CLI.
2) Arguments should use dashes instead of underscores. e.g. `--remote-host`. Python will automatically represent that as the `remote_host` attribute on args.
3) Use multiline strings for help `"""like this"""`

Reply:
Done.